sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalsShouldStartOnNewLine.cs

```diff
@@ -44,27 +44,24 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
             {
                 var ifKeyword = ((IfStatementSyntax)c.Node).IfKeyword;
+                SyntaxToken previousTokenInSameLine;
 
-                var previousSameLineToken = GetPreviousSameLineTokenOrDefault(ifKeyword);
-
-                if (previousSameLineToken != null &&
-                    previousSameLineToken.Value.IsKind(SyntaxKind.CloseBraceToken))
+                if (TryGetPreviousTokenInSameLine(ifKeyword, out previousTokenInSameLine) &&
+                    previousTokenInSameLine.IsKind(SyntaxKind.CloseBraceToken))
                 {
                     c.ReportDiagnostic(Diagnostic.Create(rule,
                             ifKeyword.GetLocation(),
-                            additionalLocations: new[] { previousSameLineToken.Value.GetLocation() }
+                            additionalLocations: new[] { previousTokenInSameLine.GetLocation() }
                         ));
                 }
             },
             SyntaxKind.IfStatement);
         }
 
-        private static SyntaxToken? GetPreviousSameLineTokenOrDefault(SyntaxToken token)
+        private static bool TryGetPreviousTokenInSameLine(SyntaxToken token, out SyntaxToken previousToken)
```

Comment:
I didn't either, but I'm growing to like it (if not abused). In this case, it makes the code simpler.